yarn-project/end-to-end/src/composed/integration_proof_verification.test.ts

```diff
@@ -0,0 +1,198 @@
+import { L2Block, deployL1Contract, fileURLToPath } from '@aztec/aztec.js';
+import { BBCircuitVerifier } from '@aztec/bb-prover';
+import { Fr, HEADER_LENGTH, Proof } from '@aztec/circuits.js';
+import { type L1ContractAddresses } from '@aztec/ethereum';
+import { type Logger } from '@aztec/foundation/log';
+import { BufferReader, serializeToBuffer } from '@aztec/foundation/serialize';
+import { AvailabilityOracleAbi, RollupAbi } from '@aztec/l1-artifacts';
+
+import { type Anvil } from '@viem/anvil';
+import { readFile } from 'fs/promises';
+import { join } from 'path';
+// @ts-expect-error solc-js doesn't publish its types https://github.com/ethereum/solc-js/issues/689
+import solc from 'solc';
+import {
+  type Account,
+  type Chain,
+  type GetContractReturnType,
+  type HttpTransport,
+  type PublicClient,
+  type WalletClient,
+  getContract,
+} from 'viem';
+import { mnemonicToAccount } from 'viem/accounts';
+
+import { AGGREGATION_OBJECT_SIZE } from '../../../bb-prover/src/verification_key/verification_key_data.js';
+import { MNEMONIC } from '../fixtures/fixtures.js';
+import { getACVMConfig } from '../fixtures/get_acvm_config.js';
+import { getBBConfig } from '../fixtures/get_bb_config.js';
+import { getLogger, setupL1Contracts, startAnvil } from '../fixtures/utils.js';
+
+/**
+ * README
+ *
+ * If this test starts failing, regenerate its fixture with
+ * AZTEC_GENERATE_TEST_DATA=1 yarn workspace @aztec/end-to-end test e2e_prover
+ */
+
+describe('proof_verification', () => {
+  let proof: Proof;
+  let block: L2Block;
+  let aggregationObject: Fr[];
+  let anvil: Anvil | undefined;
+  let walletClient: WalletClient<HttpTransport, Chain, Account>;
+  let publicClient: PublicClient<HttpTransport, Chain>;
+  // eslint-disable-next-line
+  let l1ContractAddresses: L1ContractAddresses;
+  let logger: Logger;
+  let circuitVerifier: BBCircuitVerifier;
+  let bbTeardown: () => Promise<void>;
+  let acvmTeardown: () => Promise<void>;
+  let verifierContract: GetContractReturnType<any, typeof walletClient>;
+
+  beforeAll(async () => {
```

Comment:
Is it possible to reuse the e2e_prover test harness here?

Reply:
I wanted something a lighter than deploying the whole L2 for this test. I'll disable it since keeping that fixture in sync with the rest of the system would be a pain.